tracing-appender/src/rolling.rs

```diff
@@ -152,6 +153,13 @@ impl RollingFileAppender {
             .expect("initializing rolling file appender failed")
     }
 
+    /// Keep the last `n` log entries on disk.
+    ///
+    /// If no value is supplied, `RollingAppender` will not remove any files.
+    pub fn keep_last_n_logs(&mut self, n: usize) {
+        self.state.keep_last = Some(n);
+    }
```

Comment:
I think this makes more sense to add this method to the [`Builder` type](https://github.com/tokio-rs/tracing/blob/a77be16e395893a82474688e2f3f54522937e64c/tracing-appender/src/rolling/builder.rs#L30), rather than on the `RollingAppender` itself. In this case, it can be configured on the `RollingAppender` type directly, but I think it's a bit confusing for users to call some configuration methods on the `Builder` type and other methods on the `RollingAppender` after it's built. What do you think?

Reply:
I agree, however in this case, back-compatibility will be a problem, since there will be an explicit new field called `keep_last`, and the users who want to upgrade to new version should specify `None` for the last argument (`keep_last`).
Decision is yours :)

Reply:
I don't think there would be a back-compatibility problem. When a new `Builder` is constructed, the value of `keep_last` would default to `None`, and it would be set if a `Builder::keep_last_n_logs` method is called. Otherwise, it will be `None` if the user doesn't call that method. The primary reason we have a `Builder` type is that it allows us to add as many new parameters as we want without having to change existing function signatures.